infrastructure/src/main/java/org/corfudb/infrastructure/CorfuServerNode.java

```diff
@@ -67,15 +72,35 @@
      * @param serverContext Initialized Server Context.
      */
     public CorfuServerNode(@Nonnull ServerContext serverContext) {
-        this(serverContext,
-                ImmutableMap.<Class, AbstractServer>builder()
-                        .put(BaseServer.class, new BaseServer(serverContext))
-                        .put(SequencerServer.class, new SequencerServer(serverContext))
-                        .put(LayoutServer.class, new LayoutServer(serverContext))
-                        .put(LogUnitServer.class, new LogUnitServer(serverContext))
-                        .put(ManagementServer.class, new ManagementServer(serverContext))
-                        .build()
-        );
+
+        LogUnitServerConfig config = LogUnitServerConfig.parse(serverContext.getServerConfig());
+        StreamLog streamLog;
```

Comment:
This seems to be a leaky abstraction, leaking streamLogFiles into CorfuServerNode.
If this needs to be shared between multiple components it needs to be done via ServerContext.

Reply:
I don’t necessarily agree that it is a leaky abstraction. ServerContext goes against the principle of Inversion of Control and makes all the components in the system tightly coupled, which makes it hard to test by supplying the test classes as dependencies and also maintain. 
If we place a stream log in there, all other servers will have access to it as well, although they don’t need to.
We need an explicit configuration phase where we do dependency injection for every component and every server. 
I don’t think that it is good to address it in this PR, though.

Reply:
As per offline discussion, there were some concerns about ownership of a stream log, namely what happens during the stream log resets. Right now, the new instance of a stream log is actually not getting created inside of the Log Unit but rather the state of a stream log is getting cleared, while the instance is still the same. So both a Management Server and a Log Unit still share the same instance. Hence that is why heal node still works as expected and is not a problem if a client decides to do the resets as well. 

However, I am not sure if it will be the case in the future. 

One of the changes for the optimal state transfer that we proposed is to not perform a full-blown reset but rather truncate the stream log and recreate a new instance of it from the truncated log. In this case, so far I see only three approaches:

1) LogUnit is an owner of a stream log and exposes some of the stream log APIs (Without having a need to call them via RPCs), then a Management Server would have to depend on a Log Unit instance explicitly.
2) LogUnit is an owner of a stream log and does not expose the APIs, then a Management Server would have to pay RPC costs (serialization/deserialization) for every write, which we tried to avoid in the first place. 
3) StreamLog is in the ServerContext as a shared instance, and when the new instance gets created it should get atomically updated inside of the context. Log Unit and Management Server go to the context to use the latest instance.